### PR TITLE
feat(docs): add inline documentation for experimental functions

### PR DESCRIPTION
**Pull request by sanderson**
Adds documentation for functions in the `experimental` package.

### Done checklist
- [x] docs/SPEC.md updated **(N/A)**
- [x] Test cases written **(N/A)**

### CODE REVIEW

File: stdlib/experimental/experimental.flux [hunk 1]
```diff
@@ -1,23 +1,232 @@
+// Package experimental includes experimental functions and packages that
+// perform various tasks.
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:2]: Consider removing.
```suggestion
// Package experimental includes experimental functions and packages.
```

File: stdlib/experimental/experimental.flux [hunk 2]
```diff
@@ -1,23 +1,232 @@
...
+//
+// ### Experimental packages are subject to change
+// Please note that experimental packages and functions and may:
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:5]:
```suggestion
// Please note that experimental packages and functions may:
```

File: stdlib/experimental/experimental.flux [hunk 3]
```diff
@@ -28,14 +237,171 @@ builtin to : (
     ?token: string,
 ) => [A] where A: Record
 
-// An experimental version of join.
+// join joins two streams of tables on the **group key and `_time` column**.
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:240]: Did you mean to bold here?

File: stdlib/experimental/experimental.flux [hunk 4]
```diff
@@ -50,7 +416,75 @@ builtin _window : (
...
+// window groups records based on time.
+// 
+// `_start` and `_stop` columns are updated to reflect the bounds of
+// the window the row's time value is in.
+// Input tables must have `_start`, `_stop`, and `_time` columns.
+// 
+// A single input record can be placed into zero or more output tables depending
+// on the specific windowing function.
+// 
+// By default the start boundary of a window will align with the Unix epoch (zero time)
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:428]:
```suggestion
// By default the start boundary of a window will align with the Unix epoch
```
Link to epoch definition instead?

File: stdlib/experimental/experimental.flux [hunk 5]
```diff
@@ -67,59 +501,641 @@ window = (
...
+// tags: transformations,aggregates
+// 
 builtin count : (<-tables: [{T with _value: A}]) => [{T with _value: int}]
 
-// An experimental version of histogramQuantile
+// histogramQuantile approximates a quantile given a histogram with the
+// cumulative distribution of the dataset.
+// 
+// Each input table represents a single histogram.
+// Each input table represents a single histogram.
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:593]:
```suggestion
// Each input table represents a single histogram.
```

File: stdlib/experimental/experimental.flux [hunk 6]
```diff
@@ -67,59 +501,641 @@ window = (
...
+// Input tables must have two columns—a count column (`_value`) and an upper bound
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:594]:
```suggestion
// Input tables must have two columns: a count column (`_value`) and an upper bound
```

File: stdlib/experimental/experimental.flux [hunk 7]
```diff
@@ -67,59 +501,641 @@ window = (
...
+// column (`le`), and neither column can be part of the group key.
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:595]:
```suggestion
// column (`le`). Neither column can be part of the group key.
```

File: stdlib/experimental/experimental.flux [hunk 8]
```diff
@@ -67,59 +501,641 @@ window = (
...
+// If the either of the bounds used in interpolation are infinite,
+// then the other finite bound is used and no interpolation is performed.
+// 
+// The output table has the same group key as the input table.
+// The function returns the value of the specified quantile from the histogram in the
+// `_value` column and drops all columns not part of the group key.
+// 
+// ## Parameters
+// - quantile: Quantile to compute (`[0.0 - 1.0]`).
+// - minValue: assumed minimum value of the dataset. Default is `0.0`.
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:612]:
```suggestion
// - minValue: Assumed minimum value of the dataset. Default is `0.0`.
```

File: stdlib/experimental/experimental.flux [hunk 9]
```diff
@@ -67,59 +501,641 @@ window = (
...
+// 
+//   When the quantile falls below the lowest upper bound, the function
+//   interpolates values between `minValue` and the lowest upper bound.
+//   When `minValue` is equal to negative infinity, the lowest upper bound is used.
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:616]: "When" or "If"?

File: stdlib/experimental/experimental.flux [hunk 10]
```diff
@@ -67,59 +501,641 @@ window = (
...
-// An experimental version of unique
+// unique returns all records containing unique values in the `_value` column.
+// 
+// `null` is considered a unique value.
+// 
+// `experimental.unique` outputs a single table for each input table and does
+// the following:
+// 
+// - Outputs a single record for each unique value.
+// - Leaves group keys, columns, and values unmodified.
```
Comment (pierwill) [inline stdlib/experimental/experimental.flux:1065]: Maybe an ordered list?

Comment (sanderson) [reply stdlib/experimental/experimental.flux:1064]: These aren't sequential. These are just general statements that describe the behavior. I'm going to restructure this section a bit.

File: stdlib/experimental/experimental.flux (hunk 3 above)
Comment (sanderson) [reply stdlib/experimental/experimental.flux:239]: Yes, joining on the group key and time is they key to this functions performance improvements and is also the key differentiator between `experimental.join` and `join`. With `experimental.join`, the only way to control what data is joined on is by changing the group key. There is a trade-off between performance and flexibility.

**Review by pierwill — APPROVED** [review]

**Pull request merged**